econml/dowhy.py

```diff
@@ -0,0 +1,222 @@
+            Whether variables in the dataframe that are not included in the causal graph should be automatically
+            included as confounder nodes
+        control_value: optional scalar (Default=0)
+            Value of the treatment in the control group, for effect estimation
+        treatment_value: optional scalar (Default=1)
+            Value of the treatment in the treated group, for effect estimation
+        target_units: optional (Default="ate")
+            The units for which the treatment effect should be estimated.
+            This can be of three types:
+                (1) a string for common specifications of target units (namely, "ate", "att" and "atc"),
+                (2) a lambda function that can be used as an index for the data (pandas DataFrame),
+                (3) a new DataFrame that contains values of the effect_modifiers and effect will be estimated
+                 only for this new data
+        kwargs: optional
+            Other keyword arguments from fit method for CATE estimator
+
+        Returns
+        -------
+        self
+
+        """
+
+        Y, T, X, W, Z = check_input_arrays(Y, T, X, W, Z)
+
+        # create dataframe
+        n_obs = Y.shape[0]
+        Y, T, X, W, Z = reshape_arrays_2dim(n_obs, Y, T, X, W, Z)
+
+        # currently dowhy only support single outcome and single treatment
+        assert Y.shape[1] == 1, "Can only accept single dimensional outcome."
+        assert T.shape[1] == 1, "Can only accept single dimensional treatment."
+
+        # column names
+        if outcome_names is None:
+            outcome_names = [f"Y{i}" for i in range(Y.shape[1])]
+        if treatment_names is None:
+            treatment_names = [f"T{i}" for i in range(T.shape[1])]
+        if feature_names is None:
+            feature_names = [f"X{i}" for i in range(X.shape[1])]
+        if confounder_names is None:
+            confounder_names = [f"W{i}" for i in range(W.shape[1])]
+        if instrument_names is None:
+            instrument_names = [f"Z{i}" for i in range(Z.shape[1])]
+        column_names = outcome_names + treatment_names + feature_names + confounder_names + instrument_names
+        df = pd.DataFrame(np.hstack((Y, T, X, W, Z)), columns=column_names)
+        self.dowhy_ = CausalModel(
+            data=df,
+            treatment=treatment_names,
+            outcome=outcome_names,
+            graph=graph,
+            common_causes=feature_names + confounder_names if X.shape[1] > 0 or W.shape[1] > 0 else None,
+            instruments=instrument_names if Z.shape[1] > 0 else None,
+            effect_modifiers=feature_names if X.shape[1] > 0 else None,
+            estimand_type=estimand_type,
+            proceed_when_unidetifiable=proceed_when_unidentifiable,
+            missing_nodes_as_confounders=missing_nodes_as_confounders
+        )
+        self.identified_estimand_ = self.dowhy_.identify_effect(proceed_when_unidentifiable=True)
```

Comment:
Why is `proceed_when_unidentifiable` True here instead of using the parameter with the same name?

Reply:
if False, it will pop up a question ask the user whether you want to continue, since here we will do identify, estimate anyway, so we should always proceed. However, when define CausalModel, we still allow user to set this parameters, then they could call est_dowhy.identify_effect() themselves.